advertools/sitemaps.py

```diff
@@ -475,27 +475,33 @@ def sitemap_to_df(sitemap_url, max_workers=8, recursive=True):
                            case you want to explore what sitemaps are available
                            after which you can decide which ones you are
                            interested in.
+    :param str user_agent: Provide a custom User-Agent string when retrieving
+                           the sitemaps from the given URL. This can be used to
+                           prevent 403 Forbidden errors.
     :return sitemap_df: A pandas DataFrame containing all URLs, as well as
                         other tags if available (``lastmod``, ``changefreq``,
                         ``priority``, or others found in news, video, or image
                         sitemaps).
     """
+    headers = DEFAULT_HEADERS.copy()
```

Comment:
Any reason why we're making a copy and not using the same variable?

Reply:
Yes. Without the copy, if the user-agent is updated via the parameter, it will be updated in the `DEFAULT_HEADERS`. Then if we ask for another sitemap without any custom user-agent, the custom user-agent from the last run will still be used. The copy is made to keep the defaults static.

Reply:
Good point!